### PR TITLE
build: Use Hatch

### PR DESCRIPTION
**Pull request by j-s-ashley**
Implement the modern build system Hatch based on [Scientific Python Best Practices](https://learn.scientific-python.org/development/guides/packaging-simple/).

### CODE REVIEW

**Comment by j-s-ashley** [comment]
@henryiii and @matthewfeickert, this is failing all checks as-is, but [passes](https://github.com/j-s-ashley/EnergyFlow/actions/runs/9959319572) all but the Windows tests when rebased onto #46 (and fails Windows in the same way). I think it's good to review!

**Comment by j-s-ashley** [comment]
> @henryiii and @matthewfeickert, this is failing checks as-is, but [passes](https://github.com/j-s-ashley/EnergyFlow/actions/runs/9959319572) all but the Windows tests when merged with #46 ([here](https://github.com/j-s-ashley/EnergyFlow/tree/b0bbe4beb039339651e438d29136ddade863defb)'s the merged branch) (and fails Windows in the same way). I think it's good to review!

Rebasing onto #49 shows all checks passing @henryiii and @matthewfeickert. This is definitely ready for review.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    "pot >= 0.8.0",
+    "pytest",
+    "scikit-learn",
+    "tensorflow >= 2.5.0",
+]
+
+[project.urls]
+Homepage = "https://energyflow.network"
+Issues = "https://github.com/pkomiske/EnergyFlow/issues"
+"Source Code" = "https://github.com/pkomiske/EnergyFlow"
```
Comment (matthewfeickert) [inline pyproject.toml:105]:
```suggestion
Issues = "https://github.com/thaler-lab/EnergyFlow/issues"
"Source Code" = "https://github.com/thaler-lab/EnergyFlow"
```

Just updating the metadata for the new URL

**Review by matthewfeickert — COMMENTED** [review]
@j-s-ashley LGTM beyond the update to use https://github.com/thaler-lab/ in the metadata. :+1:

File: pyproject.toml (hunk 1 above)
Comment (j-s-ashley) [reply pyproject.toml:105]: Great catch. Thank you!

**Review by matthewfeickert — APPROVED** [review]

**Pull request merged**